posthog/management/commands/fix_person_distinct_ids_after_delete.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+
+import structlog
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from posthog.client import sync_execute
+from posthog.models.person import PersonDistinctId
+from posthog.models.person.util import create_person_distinct_id
+
+logger = structlog.get_logger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Fix state for person distinct IDs in ClickHouse after person deletion and id re-use for a single team"
+
+    def add_arguments(self, parser):
+        parser.add_argument("--team-id", default=None, type=int, help="Specify a team to fix data for.")
+        parser.add_argument("--new-version", default=2500, type=int, help="New version value to use")
+        parser.add_argument("--live-run", action="store_true", help="Run changes, default is dry-run")
+
+    def handle(self, *args, **options):
+        run(options)
+
+
+def run(options):
+    live_run = options["live_run"]
+
+    if not options["team_id"]:
+        logger.error("You must specify --team-id to run this script")
+        exit(1)
+
+    team_id = options["team_id"]
+    version = options["new_version"]
+
+    distinct_ids = get_distinct_ids_tied_to_deleted_persons(team_id)
+
+    for distinct_id in distinct_ids:
+        # this can throw but this script can safely be re-run as
+        # updated distinct_ids won't show up in the search anymore
+        # since they no longer belong to deleted persons
+        # it's safer to throw and exit if anything went wrong
+        update_distinct_id(distinct_id, version, team_id, live_run)
+
+
+def get_distinct_ids_tied_to_deleted_persons(team_id: int) -> List[str]:
+    # find distinct_ids where the person is set to be deleted
+    rows = sync_execute(
+        """
+            SELECT distinct_id FROM (
```

Comment:
we've previously written incorrect sql - can we test failure modes a bit? your test covers IDs linking to a person that aren't also linked to a deleted one which is 👍  

can we test for instance what happens to IDs linking to deleted persons that weren't reused?